clients/admin-ui/src/features/privacy-requests/RequestTable.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  Table,
+  Text,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Button,
+  Flex,
+} from '@fidesui/react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import debounce from 'lodash.debounce';
+
+import { PrivacyRequest } from './types';
+import {
+  selectPrivacyRequestFilters,
+  useGetAllPrivacyRequestsQuery,
+  setPage,
+} from './privacy-requests.slice';
+
+import RequestRow from './RequestRow';
+
+interface RequestTableProps {
+  requests?: PrivacyRequest[];
+}
+
+const useRequestTable = () => {
+  const dispatch = useDispatch();
+  const filters = useSelector(selectPrivacyRequestFilters);
+  const [cachedFilters, setCachedFilters] = useState(filters);
+  const updateCachedFilters = useRef(
+    debounce((updatedFilters) => setCachedFilters(updatedFilters), 250)
+  );
+  useEffect(() => {
+    updateCachedFilters.current(filters);
+  }, [setCachedFilters, filters]);
+
+  const handlePreviousPage = () => {
+    dispatch(setPage(filters.page - 1));
+  };
+
+  const handleNextPage = () => {
+    dispatch(setPage(filters.page + 1));
+  };
+
+  const { data, isLoading } = useGetAllPrivacyRequestsQuery(cachedFilters);
+  const { items: requests, total } = data || { items: [], total: 0 };
+  return {
+    ...filters,
+    isLoading,
+    requests,
+    total,
+    handleNextPage,
+    handlePreviousPage,
+  };
+};
+
+const RequestTable: React.FC<RequestTableProps> = () => {
+  const { requests, total, page, size, handleNextPage, handlePreviousPage } =
+    useRequestTable();
+  return (
+    <>
+      <Table size="sm">
+        <Thead>
+          <Tr>
+            <Th pl={0}>Status</Th>
+            <Th>Policy Name</Th>
+            <Th>Subject Identity</Th>
+            <Th>Time Received</Th>
+            <Th>Reviewed By</Th>
+            <Th>Request ID</Th>
+            <Th />
+          </Tr>
+        </Thead>
+        <Tbody>
+          {requests.map((request) => (
+            <RequestRow request={request} key={request.id} />
+          ))}
+        </Tbody>
+      </Table>
+      <Flex justifyContent="space-between" mt={6}>
+        <Text fontSize="xs" color="gray.600">
+          Showing {(page - 1) * size + 1} to {Math.min(total, page * size)} of{' '}
```

Comment:
Are we checking that what we're passing to Math.min are always numbers so we don't run into the dreaded NaN 😱